app-backend/batch/src/main/scala/groundwork/CreateTaskGrid.scala

```diff
@@ -0,0 +1,96 @@
+package com.rasterfoundry.batch.groundwork
+
+import com.rasterfoundry.batch.Job
+import com.rasterfoundry.database.util.RFTransactor
+import com.rasterfoundry.database.{
+  AnnotationProjectDao,
+  ProjectDao,
+  TaskDao,
+  UserDao
+}
+import com.rasterfoundry.datamodel.{Task, TaskStatus}
+
+import cats.data.OptionT
+import cats.effect.{IO, LiftIO}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import doobie.implicits._
+import doobie.{ConnectionIO, Transactor}
+
+import java.util.UUID
+
+class CreateTaskGrid(
+    annotationProjectId: UUID,
+    taskSizeMeters: Double,
+    xa: Transactor[IO]
+) extends LazyLogging {
+
+  private def debug(s: String): ConnectionIO[Unit] =
+    LiftIO[ConnectionIO].liftIO(
+      IO { logger.debug(s) }
+    )
+
+  def run(): IO[Unit] =
+    (for {
+      annotationProject <- OptionT {
+        AnnotationProjectDao.getProjectById(annotationProjectId)
+      }
+      _ <- OptionT.liftF {
+        debug(s"Got annotation project ${annotationProject.name}")
+      }
+      owner <- OptionT {
+        UserDao.getUserById(annotationProject.createdBy)
+      }
+      footprint <- OptionT {
+        annotationProject.projectId traverse { projectId =>
+          ProjectDao.getFootprint(projectId)
+        }
+      } <* OptionT.liftF {
```

Comment:
`<*` here instead of `_ <- ...` because scala was getting mad about an unused variable -- it's a thing that happens sometimes with mixes of `_ <- ...` and assignments in for comprehensions as a result of the desugaring, I think